drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/test/java/org/drools/workbench/screens/guided/dtable/backend/server/indexing/IndexGuidedDecisionTableGraphFileTest.java

```diff
@@ -0,0 +1,94 @@
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.guided.dtable.backend.server.indexing;
+
+import java.io.IOException;
+
+import org.apache.lucene.search.Query;
+import org.drools.workbench.screens.guided.dtable.backend.server.GuidedDTGraphXMLPersistence;
+import org.drools.workbench.screens.guided.dtable.model.GuidedDecisionTableEditorGraphModel;
+import org.drools.workbench.screens.guided.dtable.type.GuidedDTableGraphResourceTypeDefinition;
+import org.junit.Test;
+import org.kie.workbench.common.services.refactoring.backend.server.BaseIndexingTest;
+import org.kie.workbench.common.services.refactoring.backend.server.TestIndexer;
+import org.kie.workbench.common.services.refactoring.backend.server.query.builder.SingleTermQueryBuilder;
+import org.kie.workbench.common.services.refactoring.model.index.terms.valueterms.ValueSharedPartIndexTerm;
+import org.kie.workbench.common.services.refactoring.service.PartType;
+import org.uberfire.backend.server.util.Paths;
+import org.uberfire.ext.metadata.engine.Index;
+import org.uberfire.java.nio.file.Path;
+
+public class IndexGuidedDecisionTableGraphFileTest extends BaseIndexingTest<GuidedDTableGraphResourceTypeDefinition> {
+
+    @Test
+    public void testIndexGuidedDecisionTableGraphEntries() throws IOException, InterruptedException {
+        //Add test files
+        final Path path = basePath.resolve("dtable-graph1.gdst-set");
+        final Path dtable1Path = basePath.resolve("dtable1.gdst");
+        final Path dtable2Path = basePath.resolve("dtable2.gdst");
+        final org.uberfire.backend.vfs.Path vfsDtable1Path = Paths.convert(dtable1Path);
+        final org.uberfire.backend.vfs.Path vfsDtable2Path = Paths.convert(dtable2Path);
+
+        final GuidedDecisionTableEditorGraphModel model = new GuidedDecisionTableEditorGraphModel();
+        model.getEntries().add(new GuidedDecisionTableEditorGraphModel.GuidedDecisionTableGraphEntry(vfsDtable1Path,
+                                                                                                     vfsDtable1Path));
+        model.getEntries().add(new GuidedDecisionTableEditorGraphModel.GuidedDecisionTableGraphEntry(vfsDtable2Path,
+                                                                                                     vfsDtable2Path));
+        final String xml = GuidedDTGraphXMLPersistence.getInstance().marshal(model);
+        ioService().write(path,
+                          xml);
+
+        Thread.sleep(5000); //wait for events to be consumed from jgit -> (notify changes -> watcher -> index) -> lucene index
```

Comment:
There's (probably) 100's of "indexing" tests all sleeping... however there is (now) support in Uberfire to remove the waits and instead use Byteman to signal when indexing is complete. Uberfire has been refactored to use the revised mechanism.. all other tests in all other repositories have not (yet). Let me look at this one.. perhaps it can be first!